### ISSUE TITLE
geom_abline 3.3.4 odd y scale

### ISSUE THREAD

**Issue by arnaud-feldmann** [tidyverse/ggplot2#4514]
Hi,

I am aware of #4024 and #3741 . Through, the abline modification in 3.3.4 has had some strange results for me, maintening my modest package disaggR and in particular its autoplot method for in_scatter. I can't decide if it's a feature or a bug.

```
object <- in_scatter(twoStepsBenchmark(turnover,construction))
n <- nrow(object)
df <- data.frame(Time=as.numeric(time(object))[-1L],
                 `High-frequency serie` = object[-n,2L],
                 `Low-frequency serie` = object[-n,1L],
                 xend = object[-1L,2L],
                 yend = object[-1L,1L], check.names = FALSE)

ggplot() +
  geom_abline(intercept = attr(object,"coefficients")["constant"],
                       slope = attr(object,"coefficients")[names(attr(object,"coefficients")) != "constant"],
                       lty = "solid", colour = "red", size = 1)  +
  geom_segment(data=df,
               aes(x = `High-frequency serie`, y = `Low-frequency serie`,
                   xend = xend, yend = yend,
                   colour = Time),
               arrow=arrow(angle = 15,
                           ends = "last",
                           type = "closed",
                           length = unit(0.1,"inches")),
               na.rm = TRUE)
```
The default y limits seems to be very odd to me
![image](https://user-images.githubusercontent.com/60670870/122256336-410f3580-cecf-11eb-8520-103fed31aa82.png)

on 3.3.3 the behavior was as expected
![image](https://user-images.githubusercontent.com/60670870/122256988-da3e4c00-cecf-11eb-8215-ff419adea645.png)

**Comment by clauswilke** [comment]
Could you please provide a reproducible example that depends on nothing but base R and ggplot2? And please simplify the example so all extraneous information is removed? This will make it easier to reason about the issue. Also, if you can, please use `geom_point()` instead of `geom_segment()`. However, if the problem only shows up for `geom_segment()` that would be good to know also.

**Comment by arnaud-feldmann** [comment]
Sorry,

Here is a very simpler example.

Thanks a lot for your work.

Arnaud

```
library(ggplot2)
df <- data.frame(X = c(670.964799810616, 738.482178701357, 774.152678327829,
                       816.022561905182, 877.503579439891, 933.507292448021),
                 Y = c(138.4, 143.9, 147.3, 155.9, 169.3, 181.4))
ggplot() +
  geom_abline(intercept = 44.28163,
              slope = 0.1410183)  +
  geom_point(data=df,
             aes(x = X, y = Y))
```

**ggplot2 3.3.3** :
![image](https://user-images.githubusercontent.com/60670870/122273874-3c07b180-cee2-11eb-825d-df094d8b84d5.png)

**ggplot2 3.3.4** :
![image](https://user-images.githubusercontent.com/60670870/122272803-0c0bde80-cee1-11eb-984d-2e44c5d8f404.png)

**Comment by clauswilke** [comment]
Looks like `geom_abline()` extends the line all the way to x = 0, no matter how far that is from the data. That definitely doesn't look like intended behavior to me.

``` r
library(ggplot2)
library(patchwork)

df <- data.frame(
  x = c(10000, 10001, 10002),
  y = c(500, 502, 501)
)

p <- ggplot(df) +
  geom_abline(
    intercept = -9500,
    slope = 1
  )  +
  geom_point(aes(x, y))

p | p + xlim(0, 10000)
#> Warning: Removed 2 rows containing missing values (geom_point).
```

![](https://i.imgur.com/Oz6T7f7.png)

<sup>Created on 2021-06-16 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Comment by ptoche** [comment]
Is it related to this? where the range of the line was extended to fix another issue:

https://github.com/tidyverse/ggplot2/pull/4422/commits/e127cc9c4184d5c9055ae367cff1e6a9ff913847

**Comment by arnaud-feldmann** [comment]
after having tried with and without, it is actually linked to that commit 041e287674da8e358676bf86945e4054491e47f9

**Comment by clauswilke** [comment]
Yes, in hindsight, this was a problematic change. We want the intercept to get transformed but we don't want it to enter the limit calculations. We may need to separate out those two functions, or let the geoms/stats declare which aesthetics should be used for each of these calculations. The latter approach might be cleaner but is more work.

**Comment by thomasp85** [comment]
I'm leaning on actually rolling this change back and doing a quick patch release.. thoughts, @clauswilke and @yutannihilation ?

**Comment by clauswilke** [comment]
I agree. The newly introduced bug is more problematic than the bug that was fixed, in my opinion.

**Comment by andrefaure** [comment]
I am also having trouble with geom_abline 3.3.4, noticed this issue and think it could be related?
https://stackoverflow.com/questions/68068628/gglot2geom-abline-fails-with-default-settings-in-log-log-plot-ggplot2-v3-3-4

**Comment by yutannihilation** [comment]
It sounds good to me to roll the change back. Also, I think the fix for the `ggsave()` issues (i.e., #4524 and #4521) might be worth including in the patch release.

**Comment by thomasp85** [comment]
Agreed on the ggsave issues — was planning on including that as well. Will work on this today

**Comment by yutannihilation** [comment]
Thanks. For #4521, my pr #4523 should fix the issue, I just need to confirm that with my old Windows. But feel free to take over :)

**Comment by thomasp85** [comment]
#4525

**Comment by arnaud-feldmann** [comment]
Hi,

Just to confirm that, as for me, ggplot2 3.3.5 solved the issue.

**Comment by billdenney** [comment]
The fix for 3.3.5 only partly fixes the issue for me (at least, I believe that the issue below is the same as the initially-reported issue here).  Specifically, when adding one log10 scale, the abline disappears now, but adding both puts it back:

``` r
library(ggplot2)

p <-
  ggplot(data.frame(x=1, y=1), aes(x=x, y=y)) +
  geom_point() +
  geom_abline(slope=1, intercept=0)

# basic plot shows up as expected
p
```

![](https://i.imgur.com/tVCnYnv.png)

``` r
# adding scale_x_log10() drops the line which is not expected
p + scale_x_log10()
```

![](https://i.imgur.com/03iIXnh.png)

``` r
# adding scale_y_log10() drops the line which is not expected
p + scale_y_log10()
```

![](https://i.imgur.com/WJ1vLn9.png)

``` r
# adding both log10 scales puts the line back as desired
p + scale_x_log10() + scale_y_log10()
```

![](https://i.imgur.com/LIsfvua.png)

<sup>Created on 2021-07-14 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
